### ISSUE TITLE
Make clippy less unhappy

### ISSUE THREAD

**Issue by jpgoldberg** [opendp/opendp#489]
This is less of a bug than of a tidiness issue.

**Describe the bug**
Running clippy (rust linter) on the rust code yields a bunch of minor warnings. None of the warnings represent real problems, but cleaning those up will make it easier to spot future problems earlier

**To Reproduce**

```
% cd rust
% cargo clippy
```
## Fixing

Looking over what clippy complains about, the fixes are easy and uncontroversial. I am happy to create a branch and submit a PR for those, but there is also a (set of) warnings from code generated from `impl_exact_int_cast_int_float!` in `cast.rs`.

```rust
macro_rules! impl_exact_int_cast_int_float {
    ($int:ty, $float:ty) => (impl ExactIntCast<$int> for $float {
        fn exact_int_cast(v_int: $int) -> Fallible<Self> {
            let v_float = v_int as $float;
            if <$float>::MIN_CONSECUTIVE > v_float || <$float>::MAX_CONSECUTIVE < v_float {
```
Clippy wants (see [manual_range_contains](https://rust-lang.github.io/rust-clippy/master/index.html#manual_range_contains)) us to use `.contains()` for checking whether a value is within a range.

I suspect that this is straight forward under the assumption that I am not too mistaken about when types are checked and enforced wrt to macro expansion. Indeed, the intent of the macro illustrates that I am not too mistake.  But I've never played with Rust macros before. (There was a time when I wrote M4 macros for sendmail.cf configurations, but that felt like several lifetimes ago.)

**Comment by Shoeboxam** [comment]
Usually I'd just say "go for it!", but we have a pretty long backlog of PR's right now, and clippy fixes usually cause conflicts. In order to avoid rebase conflicts throughout the PR backlog, would you mind branching off of `486-feature-docs` and then request to merge into `486-feature-docs`? 

Thanks for taking a look at this! And I agree with clippy, `.contains()` would be clearer. In the case of that macro, it may be more readable if you substitute `<$float>` with `Self`. You're also welcome to join the Slack (link is in the README.md) if you'd like to chat directly.

**Comment by jpgoldberg** [comment]
Just an update.

1. I hadn't seen the advice to work from 486-feature-docs before I started. I will try merging that into my branch, to help avoid conflicts.
2. The parts that I thought would be easy turned out to be hard and the parts I thought would be hard turned out to be easy.
3. It turns out that there are lots of instances where the clippy is giving some bad advice. I don't like using `#[allow()]` just to shut it up, but I will do those when there is a good reason to not do things clippy's way. I attempted to comment those reasons.
4. There are two cases where I made no decision about doing what clippy wanted or adding an `#[allow(...)]`
5. There are changes I made which I suspect should be subject to extremely careful review. I tried to be conservative, particularly around code that I didn't fully understand. But I will try to draw attention to things that may need special attention.
6. This was a lot of fun. Some clippy warnings were new to me and I learned a lot. I also became much more familiar with the code.
